### PR TITLE
Update `poetry.lock` and `pyproject.toml`, add CI check on `poetry.lock`

### PR DESCRIPTION
**Pull request by marcofavoritobi**
This PR simply refreshes the dependencies by running `poetry update`. Then, the dependency sections of `tox.ini` are synchronized with the new content of `poetry.lock`.

Also `pyproject.toml` is synchronized with the new `poetry.lock`, although the update required a couple of cyclic iteration with `poetry lock` until convergence.

The new version of `numpy` had some breaking changes which affected the correct behaviour of the library. The errors have been fixed [3eb5350](https://github.com/bancaditalia/black-it/pull/72/commits/3eb535076a61de53e074c4c5eac52c7bd95794e3).

Moreover, we added a CI step that checks whether `poetry.lock` is up-to-date.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -24,7 +24,14 @@ jobs:
...
     - name: Install dependencies
-      run:  pip install tox
+      run:  pip install poetry tox
+    - name: Check Poetry lock file
+      run: |
+        poetry check
+        echo "Info: 'poetry check' passed. Now trying to update the lock file."
+        poetry lock
+        git diff --exit-code poetry.lock || (echo "Error: poetry.lock file has changed. Please update it and commit the changes." && exit 1)
+        echo "Info: poetry.lock file is up-to-date with pyproject.toml."
```
Comment (marcofavoritobi) [inline .github/workflows/lint.yml:34]: @muxator is it OK for you?

**Review by AldoGl — APPROVED** [review]
Needed update, thanks

File: .github/workflows/lint.yml (hunk 1 above)
Comment (muxator) [reply .github/workflows/lint.yml:34]: Talked offline, we'll replace it with a `poetry check --lock`.

File: Makefile [hunk 2]
```diff
@@ -68,6 +68,10 @@ lint-all: black ruff static bandit safety vulture darglint ## run all linters
 .PHONY: lint-all-files
 lint-all-files: black-files ruff-files static-files bandit-files vulture-files darglint-files ## run all linters for specific files (specified with files="file1 file2 somedir ...")
 
+.PHONY: poetry-lock-check
+poetry-lock-check: ## check if poetry.lock is up to date
```
Comment (muxator) [inline Makefile:72]: check if poetry.lock "is up to date" -> "is consistent with pyproject.toml"

Comment (muxator) [reply Makefile:72]: Done in f64b31306d9e (force pushed)

**Pull request merged**